diff -U0 pycups-1.9.47/ChangeLog.request-readio pycups-1.9.47/ChangeLog
diff -up pycups-1.9.47/cupsipp.c.request-readio pycups-1.9.47/cupsipp.c
--- pycups-1.9.47/cupsipp.c.request-readio	2009-08-18 11:56:03.000000000 +0100
+++ pycups-1.9.47/cupsipp.c	2010-01-06 15:54:37.978283521 +0000
@@ -1,6 +1,6 @@
 /*
  * cups - Python bindings for CUPS
- * Copyright (C) 2002, 2005, 2006, 2007, 2008, 2009  Red Hat, Inc.
+ * Copyright (C) 2002, 2005, 2006, 2007, 2008, 2009, 2010  Red Hat, Inc.
  * Author: Tim Waugh <twaugh@redhat.com>
  *
  * This program is free software; you can redistribute it and/or modify
@@ -421,7 +421,7 @@ IPPRequest_getAttributes (IPPRequest *se
 	if (!value) {
 	  Py_DECREF (values);
 	  values = NULL;
-	  break;
+	  continue;
 	}
 
 	debugprintf ("\n");
