diff -up pycups-1.9.31/cupsconnection.c.job-sheets pycups-1.9.31/cupsconnection.c
--- pycups-1.9.31/cupsconnection.c.job-sheets	2007-11-22 11:01:19.000000000 +0000
+++ pycups-1.9.31/cupsconnection.c	2007-11-22 13:27:43.000000000 +0000
@@ -2042,7 +2042,7 @@ Connection_getPrinterAttributes (Connect
 	startobj = PyObj_from_UTF8 (start);
 	endobj = PyObj_from_UTF8 (end);
 	PyDict_SetItemString (ret, "job-sheets-default",
-			      Py_BuildValue ("(ss)", startobj, endobj));
+			      Py_BuildValue ("(OO)", startobj, endobj));
 	Py_DECREF (startobj);
 	Py_DECREF (endobj);
 	continue;
diff -U0 pycups-1.9.31/ChangeLog.job-sheets pycups-1.9.31/ChangeLog
--- pycups-1.9.31/ChangeLog.job-sheets	2007-11-22 11:04:25.000000000 +0000
+++ pycups-1.9.31/ChangeLog	2007-11-22 13:27:43.000000000 +0000
@@ -2,0 +3,5 @@
+	* cupsconnection.c (Connection_getPrinterAttributes): Fixed
+	job-sheets-default tuple.
+
+2007-11-22  Tim Waugh  <twaugh@redhat.com>
+
