diff -up system-config-printer-1.0.7/authconn.py.forbidden system-config-printer-1.0.7/authconn.py
--- system-config-printer-1.0.7/authconn.py.forbidden	2008-08-29 11:32:49.000000000 +0100
+++ system-config-printer-1.0.7/authconn.py	2008-08-30 10:42:09.000000000 +0100
@@ -164,7 +164,8 @@ class Connection:
                         raise cups.IPPError (cups.IPP_NOT_AUTHORIZED, '')
                 break
             except cups.IPPError, (e, m):
-                if not self._cancel and e == cups.IPP_NOT_AUTHORIZED:
+                if not self._cancel and (e == cups.IPP_NOT_AUTHORIZED or
+                                         e == cups.IPP_FORBIDDEN):
                     self._failed ()
                 else:
                     raise
diff -U0 system-config-printer-1.0.7/ChangeLog.forbidden system-config-printer-1.0.7/ChangeLog
--- system-config-printer-1.0.7/ChangeLog.forbidden	2008-08-29 11:32:49.000000000 +0100
+++ system-config-printer-1.0.7/ChangeLog	2008-08-30 10:42:09.000000000 +0100
@@ -0,0 +1,5 @@
+2008-08-30  Tim Waugh  <twaugh@redhat.com>
+
+	* authconn.py (Connection._authloop): Handle IPP_FORBIDDEN (bug
+	#460670).
+
