diff -up system-config-printer-1.1.13/jobviewer.py.jobs-window-visibility system-config-printer-1.1.13/jobviewer.py
--- system-config-printer-1.1.13/jobviewer.py.jobs-window-visibility	2009-09-04 10:35:48.000000000 +0100
+++ system-config-printer-1.1.13/jobviewer.py	2009-10-28 15:47:52.796015837 +0000
@@ -378,6 +378,7 @@ class JobViewer (GtkGUI, monitor.Watcher
     def on_delete_event(self, *args):
         if self.trayicon or not self.loop:
             self.JobsWindow.hide ()
+            self.JobsWindow.set_data ('visible', False)
             if not self.loop:
                 # Being run from main app, not applet
                 self.cleanup ()
