diff -up system-config-printer-1.4.5/jobviewer.py.retrieve system-config-printer-1.4.5/jobviewer.py
--- system-config-printer-1.4.5/jobviewer.py.retrieve	2014-07-04 16:21:48.000000000 +0100
+++ system-config-printer-1.4.5/jobviewer.py	2014-07-14 11:04:27.392705510 +0100
@@ -1424,14 +1424,19 @@ class JobViewer (GtkGUI):
                                      host=self.host,
                                      port=self.port,
                                      encryption=self.encryption)
-        except RuntimeError:
+        except RuntimeError as e:
+            print e
             return
 
         for jobid in self.jobids:
             try:
                 attrs=c.getJobAttributes(jobid)
                 printer_uri=attrs['job-printer-uri']
-                document_count=attrs.get ('document-count', 0)
+                try:
+                    document_count = attrs['number-of-documents']
+                except KeyError:
+                    document_count = attrs.get ('document-count', 0)
+
                 for document_number in range(1, document_count+1):
                     document=c.getDocument(printer_uri, jobid, document_number)
                     tempfile = document.get('file')
