diff -up pycups-1.9.49/Makefile.pycups-build pycups-1.9.49/Makefile
--- pycups-1.9.49/Makefile.pycups-build	2010-03-22 12:46:31.351447572 +0000
+++ pycups-1.9.49/Makefile	2010-03-22 12:46:41.155449391 +0000
@@ -31,7 +31,7 @@ dist:
 install:
 	ROOT= ; \
 	if [ -n "$$DESTDIR" ]; then ROOT="--root $$DESTDIR"; fi; \
-	python setup.py install $$ROOT
+	CFLAGS=-DVERSION=\\\"$(VERSION)\\\" python setup.py install $$ROOT
 
 .PHONY: doc clean dist install
 
