diff -up system-config-printer-1.1.90/system-config-printer.py.copy-printer system-config-printer-1.1.90/system-config-printer.py
--- system-config-printer-1.1.90/system-config-printer.py.copy-printer	2010-01-19 13:06:19.000000000 +0000
+++ system-config-printer-1.1.90/system-config-printer.py	2010-01-21 14:45:33.668150157 +0000
@@ -3058,7 +3058,7 @@ class GUI(GtkGUI, monitor.Watcher):
                 self.cups._end_operation ()
                 return
 
-        if self.copy_printer (new_name):
+        if self.duplicate_printer (new_name):
             # Failure.
             self.monitor.update ()
 
