diff -up system-config-printer-1.1.13/system-config-printer.py.network-model system-config-printer-1.1.13/system-config-printer.py
--- system-config-printer-1.1.13/system-config-printer.py.network-model	2009-09-30 16:55:46.115760592 +0100
+++ system-config-printer-1.1.13/system-config-printer.py	2009-09-30 16:55:52.545635754 +0100
@@ -4217,6 +4217,12 @@ class NewPrinterGUI(GtkGUI):
                 if not self.install_hplip_plugin(self.device.uri):
                     self.on_NPCancel(None)
                     return
+
+                if not devid and self.device.type in ["socket", "lpd", "ipp"]:
+                    # This is a network printer whose model we don't yet know.
+                    # Try to discover it.
+                    self.getNetworkPrinterMakeModel ()
+
                 uri = self.device.uri
                 if uri and uri.startswith ("smb://"):
                     uri = SMBURI (uri=uri[6:]).sanitize_uri ()
