diff -up system-config-printer-1.1.8/authconn.py.incorrect-auth system-config-printer-1.1.8/authconn.py
--- system-config-printer-1.1.8/authconn.py.incorrect-auth	2009-06-18 16:45:49.000000000 +0100
+++ system-config-printer-1.1.8/authconn.py	2009-06-25 17:51:23.943329108 +0100
@@ -389,6 +389,8 @@ class Connection:
                       _("The password may be incorrect."))
         if self._lock:
             d.connect ("response", self._on_not_authorized_dialog_response)
+            d.show_all ()
+            d.show_now ()
             gtk.gdk.threads_leave ()
         else:
             d.run ()
