diff -up system-config-printer-1.2.96/system-config-printer.py.ungrab-focus system-config-printer-1.2.96/system-config-printer.py
--- system-config-printer-1.2.96/system-config-printer.py.ungrab-focus	2011-01-14 17:17:32.000000000 +0100
+++ system-config-printer-1.2.96/system-config-printer.py	2011-01-21 13:07:54.000000000 +0100
@@ -1404,7 +1404,6 @@ class GUI(GtkGUI):
         self.rename_sigids = ids
         self.rename_entry_sigid = None
         self.dests_iconview.set_cursor (path, cell, start_editing=True)
-        self.dests_iconview.grab_focus ()
 
     def printer_name_edit_start (self, cell, editable, path):
         debugprint ("editing-started")
