diff -up system-config-printer-1.1.13/system-config-printer.py.gpk-traceback system-config-printer-1.1.13/system-config-printer.py
--- system-config-printer-1.1.13/system-config-printer.py.gpk-traceback	2009-11-23 14:11:10.000000000 +0100
+++ system-config-printer-1.1.13/system-config-printer.py	2009-11-23 14:11:53.000000000 +0100
@@ -4316,9 +4316,9 @@ class NewPrinterGUI(GtkGUI):
                         debugprint ("foomatic-db-ppds already installed")
                     except OSError:
                         debugprint ("foomatic-db-ppds not yet installed")
-                        pk = installpackage.PackageKit ()
                         pid = None
                         try:
+                            pk = installpackage.PackageKit ()
                             xid = self.mainapp.PrintersWindow.window.xid
                             pk.InstallPackageName (xid, 0, "foomatic-db-ppds")
                         except:
