diff -up system-config-printer-1.1.13/PhysicalDevice.py.physdev-traceback system-config-printer-1.1.13/PhysicalDevice.py
--- system-config-printer-1.1.13/PhysicalDevice.py.physdev-traceback	2009-09-01 12:08:33.000000000 +0100
+++ system-config-printer-1.1.13/PhysicalDevice.py	2009-09-30 16:52:24.486760364 +0100
@@ -48,6 +48,8 @@ class PhysicalDevice:
                     hostport = ipparam[3:]
                 else:
                     return None
+            else:
+                return None
         else:
             (hostport, rest) = urllib.splithost (rest)
             if hostport == None:
