diff -up system-config-printer-1.2.0/asyncpk1.py.jobcancel-compat system-config-printer-1.2.0/asyncpk1.py
--- system-config-printer-1.2.0/asyncpk1.py.jobcancel-compat	2010-04-12 12:09:40.000000000 +0100
+++ system-config-printer-1.2.0/asyncpk1.py	2010-04-12 12:12:03.562024987 +0100
@@ -441,11 +441,10 @@ class PK1Connection:
         (use_pycups, reply_handler, error_handler,
          tup) = self._args_kwds_to_tuple ([int],
-                                          [(None, None),
-                                           (None, False)], # purge_job
+                                          [(None, None)],
                                           args, kwds)
 
         self._call_with_pk (use_pycups,
-                            'JobCancelPurge', tup, reply_handler, error_handler,
+                            'JobCancel', tup, reply_handler, error_handler,
                             self._nothing_to_unpack,
                             self._conn.cancelJob, args, kwds)
 
