diff -up system-config-printer-1.1.8/cupshelpers/cupshelpers.py.gutenprint system-config-printer-1.1.8/cupshelpers/cupshelpers.py
--- system-config-printer-1.1.8/cupshelpers/cupshelpers.py.gutenprint	2009-06-17 18:39:51.000000000 +0100
+++ system-config-printer-1.1.8/cupshelpers/cupshelpers.py	2009-07-03 18:09:03.598995561 +0100
@@ -736,8 +736,10 @@ def missingPackagesAndExecutables(ppd):
             # IJS servers (used by foomatic)
             'hpijs': 'hpijs',
             'ijsgutenprint.5.0': 'gutenprint',
+            'ijsgutenprint.5.2': 'gutenprint',
             # CUPS filters
             'rastertogutenprint.5.0': 'gutenprint-cups',
+            'rastertogutenprint.5.2': 'gutenprint-cups',
             'commandtoepson': 'gutenprint-cups',
             'commandtocanon': 'gutenprint-cups',
             }
