diff -up system-config-printer-1.1.13/cupshelpers/ppds.py.strip-zjs system-config-printer-1.1.13/cupshelpers/ppds.py
--- system-config-printer-1.1.13/cupshelpers/ppds.py.strip-zjs	2009-10-29 16:51:28.575897013 +0000
+++ system-config-printer-1.1.13/cupshelpers/ppds.py	2009-10-29 17:18:17.526021102 +0000
@@ -184,6 +184,7 @@ def ppdMakeModelSplit (ppd_make_and_mode
                    " ps3",
                    " pxl",
                    " series",
+                   " zjs",              # hpcups
                    " zxs",              # hpcups
                    " pcl3",             # hpcups
                    "_bt",
