diff -up system-config-printer-1.5.2/scp-dbus-service.py.python3 system-config-printer-1.5.2/scp-dbus-service.py
--- system-config-printer-1.5.2/scp-dbus-service.py.python3	2014-10-10 16:42:42.000000000 +0100
+++ system-config-printer-1.5.2/scp-dbus-service.py	2014-10-10 17:20:32.792518638 +0100
@@ -295,7 +295,7 @@ class GroupPhysicalDevicesRequest:
         # (in self.devices) have been modified.
         try:
             self.physdevs = []
-            for device_uri, deviceobj in self.deviceobjs.iteritems ():
+            for device_uri, deviceobj in self.deviceobjs.items ():
                 newphysicaldevice = PhysicalDevice.PhysicalDevice (deviceobj)
                 matched = False
                 try:
