diff -up system-config-printer-0.7.78/print-applet.desktop.in.0.7.x system-config-printer-0.7.78/print-applet.desktop.in
--- system-config-printer-0.7.78/print-applet.desktop.in.0.7.x	2007-11-22 15:23:59.000000000 +0000
+++ system-config-printer-0.7.78/print-applet.desktop.in	2007-11-27 12:51:17.000000000 +0000
@@ -5,6 +5,6 @@ _Comment=System tray icon for managing p
 Exec=/usr/bin/system-config-printer-applet
 Terminal=false
 Type=Application
-Icon=printer.png
+Icon=printer
 X-KDE-autostart-after=panel
 StartupNotify=false
diff -up system-config-printer-0.7.78/system-config-printer.py.0.7.x system-config-printer-0.7.78/system-config-printer.py
--- system-config-printer-0.7.78/system-config-printer.py.0.7.x	2007-11-22 15:23:59.000000000 +0000
+++ system-config-printer-0.7.78/system-config-printer.py	2007-11-27 12:51:17.000000000 +0000
@@ -3231,10 +3231,16 @@ class NewPrinterGUI(GtkGUI):
             try:
                 cups.setServer (match.group (2))
                 c = cups.Connection ()
-                attributes = c.getPrinterAttributes (match.group (4))
+                try:
+                    attributes = c.getPrinterAttributes (uri = uri)
+                except TypeError: # uri keyword introduced in pycups 1.9.32
+                    print "Fetching printer attributes by name"
+                    attributes = c.getPrinterAttributes (match.group (4))
                 verified = True
+            except cups.IPPError (e, msg):
+                print "Failed to get attributes:", e, msg
             except:
-                pass
+                nonfatalError ()
         else:
             print uri
 
@@ -3279,9 +3285,8 @@ class NewPrinterGUI(GtkGUI):
             printers = c.getPrinters ()
             classes = c.getClasses ()
             del c
-        except RuntimeError:
-            failed = True
-        except cups.IPP_Error, (e, msg):
+        except:
+            nonfatalException()
             failed = True
 
         gtk.gdk.threads_enter()
diff -U0 system-config-printer-0.7.78/ChangeLog.0.7.x system-config-printer-0.7.78/ChangeLog
--- system-config-printer-0.7.78/ChangeLog.0.7.x	2007-11-22 15:23:59.000000000 +0000
+++ system-config-printer-0.7.78/ChangeLog	2007-11-27 12:51:17.000000000 +0000
@@ -0,0 +1,14 @@
+2007-11-27  Tim Waugh  <twaugh@redhat.com>
+
+	* system-config-printer.py
+	(NewPrinterGUI.browse_ipp_queues_thread): Handle exceptions other
+	than RuntimeError and cups.IPPError (bug #252304).
+	(NewPrinterGUI.on_btnIPPVerify_clicked): Get printer attributes by
+	printer URI not name, to prevent pycups constructing a CUPS URI
+	(bug #252304).
+
+2007-11-22  Till Kamppeter  <till.kamppeter@gmail.com>
+
+	* print-applet.desktop.in: Removed the last printer.png (Ubuntu package
+	did not build with Icon=printer.png in the .desktop file).
+
