diff -U0 system-config-printer-0.9.91/ChangeLog.iconpath system-config-printer-0.9.91/ChangeLog
diff -up system-config-printer-0.9.91/system-config-printer.py.iconpath system-config-printer-0.9.91/system-config-printer.py
--- system-config-printer-0.9.91/system-config-printer.py.iconpath	2008-05-16 16:42:41.000000000 +0100
+++ system-config-printer-0.9.91/system-config-printer.py	2008-05-18 12:42:24.000000000 +0100
@@ -82,7 +82,7 @@ import gettext
 gettext.textdomain (domain)
 gtk.glade.bindtextdomain (domain)
 pkgdata = config.Paths ().get_path ('pkgdatadir')
-iconpath = os.path.join (pkgdata, '/icons/')
+iconpath = os.path.join (pkgdata, 'icons/')
 glade_file = os.path.join (pkgdata, domain + '.glade')
 sys.path.append (pkgdata)
 
