From 13c584b417496be4a99560a71bf0950cf9ec5269 Mon Sep 17 00:00:00 2001
From: Tim Waugh <twaugh@redhat.com>
Date: Fri, 23 May 2008 16:17:03 +0100
Subject: [PATCH] Removed Scanning... label when SMB browsing shows no workgroups.

---
 ChangeLog                |    4 ++++
 system-config-printer.py |    2 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 9b10f43..3d7c547 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,9 @@
 2008-05-23  Tim Waugh  <twaugh@redhat.com>
 
+	* system-config-printer.py
+	(NewPrinterGUI.browse_smb_hosts_thread): Remove the 'Scanning...'
+	label when browsing is unsuccessful.
+
 	* configure.in: Version 0.9.93.
 
 2008-05-23  Tim Waugh  <twaugh@redhat.com>
diff --git a/system-config-printer.py b/system-config-printer.py
index a793ef1..6564de8 100755
--- a/system-config-printer.py
+++ b/system-config-printer.py
@@ -3355,8 +3355,8 @@ class NewPrinterGUI(GtkGUI):
                 store.set_value (iter, 0, d['DOMAIN'])
                 store.set_value (iter, 2, d)
         else:
+            store.clear ()
             if workgroups:
-                store.clear ()
                 for workgroup in workgroups:
                     iter = store.append (None, [workgroup])
                     i = store.append (iter)
-- 
1.5.4.3

