diff -up system-config-printer-1.1.11/udev/udev-configure-printer.c.hplip-hack system-config-printer-1.1.11/udev/udev-configure-printer.c
--- system-config-printer-1.1.11/udev/udev-configure-printer.c.hplip-hack	2009-08-07 17:42:55.000000000 +0100
+++ system-config-printer-1.1.11/udev/udev-configure-printer.c	2009-08-14 15:50:32.302334316 +0100
@@ -791,8 +791,8 @@ find_matching_device_uris (struct device
 
       /* Now check the manufacturer and model names. */
       if (device_uri && this_id.mfg && this_id.mdl &&
-	  !strcmp (this_id.mfg, id->mfg) &&
-	  !strcmp (this_id.mdl, id->mdl))
+	  !strcasecmp (this_id.mfg, id->mfg) &&
+	  !strcasecmp (this_id.mdl, id->mdl))
 	{
 	  /* We've checked everything except the serial numbers.  This
 	   * is more complicated.  Some devices include a serial
