diff -up system-config-printer-1.2.96/newprinter.py.dnssd-URI system-config-printer-1.2.96/newprinter.py
--- system-config-printer-1.2.96/newprinter.py.dnssd-URI	2011-01-18 11:59:38.000000000 +0100
+++ system-config-printer-1.2.96/newprinter.py	2011-01-18 12:00:05.000000000 +0100
@@ -2156,7 +2156,7 @@ class NewPrinterGUI(GtkGUI):
             entry.set_text(new_text)
 
     def on_entNPTDevice_changed(self, ent):
-        allowed_chars = string.letters+string.digits+'_-./:'
+        allowed_chars = string.letters+string.digits+'_-./:%()'
         self.entry_changed(ent, allowed_chars)
         self.setNPButtons()
 
