diff -up system-config-printer-1.1.92/statereason.py.cupsGetPPD2 system-config-printer-1.1.92/statereason.py
--- system-config-printer-1.1.92/statereason.py.cupsGetPPD2	2010-02-18 16:17:48.000000000 +0100
+++ system-config-printer-1.1.92/statereason.py	2010-02-22 12:10:09.000000000 +0100
@@ -138,7 +138,7 @@ class StateReason:
                     f = self.connection.getPPD(self.printer)
                     self._ppd = cups.PPD (f)
                     os.unlink (f)
-                except (cups.IPPError, OSError):
+                except (cups.IPPError, RuntimeError, OSError):
                     pass
 
             reason = self.get_reason ()
