diff -up system-config-printer-1.1.13/glade/PrintersWindow.glade.center system-config-printer-1.1.13/glade/PrintersWindow.glade
--- system-config-printer-1.1.13/glade/PrintersWindow.glade.center	2009-09-04 10:35:48.000000000 +0100
+++ system-config-printer-1.1.13/glade/PrintersWindow.glade	2009-11-26 15:36:40.465178836 +0000
@@ -8,7 +8,7 @@
   <property name="visible">False</property>
   <property name="title" translatable="yes">System-Config-Printer</property>
   <property name="type">GTK_WINDOW_TOPLEVEL</property>
-  <property name="window_position">GTK_WIN_POS_NONE</property>
+  <property name="window_position">GTK_WIN_POS_CENTER</property>
   <property name="modal">False</property>
   <property name="default_width">450</property>
   <property name="default_height">250</property>
