diff -up system-config-printer-1.5.7/jobviewer.py.minorerrors system-config-printer-1.5.7/jobviewer.py
--- system-config-printer-1.5.7/jobviewer.py.minorerrors	2016-10-24 17:37:43.212104425 +0200
+++ system-config-printer-1.5.7/jobviewer.py	2016-10-24 17:52:58.886209601 +0200
@@ -25,6 +25,8 @@ import dbus
 import dbus.glib
 import dbus.service
 import threading
+import gi
+gi.require_version('Notify', '0.7')
 from gi.repository import Notify
 from gi.repository import GLib
 from gi.repository import GObject
@@ -52,6 +54,7 @@ from functools import reduce
 cups.require("1.9.47")
 
 try:
+    gi.require_version('GnomeKeyring', '1.0')
     from gi.repository import GnomeKeyring
     USE_KEYRING=True
 except ImportError:
diff -up system-config-printer-1.5.7/system-config-printer.py.minorerrors system-config-printer-1.5.7/system-config-printer.py
--- system-config-printer-1.5.7/system-config-printer.py.minorerrors	2016-10-24 17:37:59.637936671 +0200
+++ system-config-printer-1.5.7/system-config-printer.py	2016-10-24 17:48:21.537238174 +0200
@@ -27,14 +27,19 @@ import config
 import sys, os, time, re
 import _thread
 import dbus
+import gi
 try:
+    gi.require_version('Polkit', '1.0')
     from gi.repository import Polkit
 except:
     Polkit = False
 
+gi.require_version('GdkPixbuf', '2.0')
 from gi.repository import GdkPixbuf
 try:
+    gi.require_version('Gdk', '3.0')
     from gi.repository import Gdk
+    gi.require_version('Gtk', '3.0')
     from gi.repository import Gtk
     Gtk.init (sys.argv)
 except RuntimeError as e:
