diff -U0 system-config-printer-1.1.8/ChangeLog.nmblookup-failure system-config-printer-1.1.8/ChangeLog
diff -up system-config-printer-1.1.8/troubleshoot/CheckPrinterSanity.py.nmblookup-failure system-config-printer-1.1.8/troubleshoot/CheckPrinterSanity.py
--- system-config-printer-1.1.8/troubleshoot/CheckPrinterSanity.py.nmblookup-failure	2009-05-12 10:36:36.000000000 +0100
+++ system-config-printer-1.1.8/troubleshoot/CheckPrinterSanity.py	2009-06-25 17:49:39.628328710 +0100
@@ -99,7 +99,8 @@ class CheckPrinterSanity(Question):
                         if line.startswith ("querying"):
                             continue
                         spc = line.find (' ')
-                        if spc != -1:
+                        if (spc != -1 and
+                            not line[spc:].startswith (" failed ")):
                             # Remember the IP address.
                             self.answers['remote_server_name'] = line[:spc]
                             break
