diff -up system-config-printer-1.1.13/probe_printer.py.missing-import system-config-printer-1.1.13/probe_printer.py
--- system-config-printer-1.1.13/probe_printer.py.missing-import	2009-09-01 12:08:33.000000000 +0100
+++ system-config-printer-1.1.13/probe_printer.py	2009-09-22 10:33:18.358546374 +0100
@@ -20,6 +20,7 @@
 
 import cupshelpers
 from debug import *
+import errno
 import socket, time
 import gtk
 from timedops import TimedOperation
