diff -up system-config-printer-1.5.9/jobviewer.py.keyring-workaround system-config-printer-1.5.9/jobviewer.py
--- system-config-printer-1.5.9/jobviewer.py.keyring-workaround	2017-06-19 16:40:42.072825363 +0200
+++ system-config-printer-1.5.9/jobviewer.py	2017-06-19 16:42:18.250024332 +0200
@@ -57,7 +57,7 @@ try:
     gi.require_version('GnomeKeyring', '1.0')
     from gi.repository import GnomeKeyring
     USE_KEYRING=True
-except ImportError:
+except (ImportError, ValueError):
     USE_KEYRING=False
 
 import gettext
