diff -up system-config-printer-1.2.2/Makefile.am.no-epydoc system-config-printer-1.2.2/Makefile.am
--- system-config-printer-1.2.2/Makefile.am.no-epydoc	2010-05-07 15:19:03.000000000 +0100
+++ system-config-printer-1.2.2/Makefile.am	2010-05-18 12:57:39.219065881 +0100
@@ -248,7 +248,9 @@ $(man_MANS): $(top_srcdir)/man/system-co
 
 html:	$(EXPORT_MODULES)
 	rm -rf html
-	epydoc -o html --html $(EXPORT_MODULES)
+	$(MKDIR_P) html
+	pydoc -w $(EXPORT_MODULES)
+	mv $(patsubst %,%.html,$(EXPORT_MODULES)) html
 
 distcheck-hook: update-po missing-imports
 
diff -up system-config-printer-1.2.2/Makefile.in.no-epydoc system-config-printer-1.2.2/Makefile.in
--- system-config-printer-1.2.2/Makefile.in.no-epydoc	2010-05-07 15:26:09.000000000 +0100
+++ system-config-printer-1.2.2/Makefile.in	2010-05-18 12:57:39.222067064 +0100
@@ -1431,7 +1431,9 @@ $(man_MANS): $(top_srcdir)/man/system-co
 
 html:	$(EXPORT_MODULES)
 	rm -rf html
-	epydoc -o html --html $(EXPORT_MODULES)
+	$(MKDIR_P) html
+	pydoc -w $(EXPORT_MODULES)
+	mv $(patsubst %,%.html,$(EXPORT_MODULES)) html
 
 distcheck-hook: update-po missing-imports
 
diff -up system-config-printer-1.2.2/pycups-1.9.49/Makefile.no-epydoc system-config-printer-1.2.2/pycups-1.9.49/Makefile
--- system-config-printer-1.2.2/pycups-1.9.49/Makefile.no-epydoc	2010-03-10 09:45:10.000000000 +0000
+++ system-config-printer-1.2.2/pycups-1.9.49/Makefile	2010-05-18 12:57:39.226942671 +0100
@@ -14,7 +14,9 @@ cups.so: $(SOURCES)
 
 doc:	cups.so
 	rm -rf html
-	epydoc -o html --html $<
+	mkdir -p html
+	pydoc -w $(patsubst %.so,%,$<)
+	mv $(patsubst %.so,%.html,$<) html
 
 clean:
 	-rm -rf build cups.so *.pyc *~
diff -up system-config-printer-1.2.2/pysmbc-1.0.7/Makefile.no-epydoc system-config-printer-1.2.2/pysmbc-1.0.7/Makefile
--- system-config-printer-1.2.2/pysmbc-1.0.7/Makefile.no-epydoc	2010-05-18 11:59:35.000000000 +0100
+++ system-config-printer-1.2.2/pysmbc-1.0.7/Makefile	2010-05-18 12:58:28.680066521 +0100
@@ -8,7 +8,9 @@ smbc.so: $(SOURCES)
 
 doc: smbc.so
 	rm -rf html
-	epydoc -o html --html $<
+	mkdir -p html
+	pydoc -w $(patsubst %.so,%,$<)
+	mv $(patsubst %.so,%.html,$<) html
 
 clean:
 	-rm -rf build smbc.so *.pyc tests/*.pyc *~ tests/*~
