Index: po/sr@Latn.po
===================================================================
Index: po/as.po
===================================================================
Index: po/bs.po
===================================================================
Index: po/cs.po
===================================================================
Index: po/pt_BR.po
===================================================================
Index: po/mk.po
===================================================================
Index: po/es.po
===================================================================
Index: po/ko.po
===================================================================
Index: po/lo.po
===================================================================
Index: po/is.po
===================================================================
Index: po/no.po
===================================================================
Index: po/sk.po
===================================================================
Index: po/ms.po
===================================================================
Index: po/uk.po
===================================================================
Index: po/ro.po
===================================================================
Index: po/system-config-printer.pot
===================================================================
Index: po/nb.po
===================================================================
Index: po/bn.po
===================================================================
Index: po/ar.po
===================================================================
Index: po/fr.po
===================================================================
Index: po/kn.po
===================================================================
Index: po/hr.po
===================================================================
Index: po/nn.po
===================================================================
Index: po/mr.po
===================================================================
Index: po/or.po
===================================================================
Index: po/lv.po
===================================================================
Index: po/ca.po
===================================================================
Index: po/sr.po
===================================================================
Index: po/da.po
===================================================================
Index: po/bn_IN.po
===================================================================
Index: po/tr.po
===================================================================
Index: po/fa.po
===================================================================
Index: po/de.po
===================================================================
Index: po/sv.po
===================================================================
Index: po/ja.po
===================================================================
Index: po/ka.po
===================================================================
Index: po/zh_TW.po
===================================================================
Index: po/fi.po
===================================================================
Index: po/pa.po
===================================================================
Index: po/hi.po
===================================================================
Index: po/zh_CN.po
===================================================================
Index: po/ta.po
===================================================================
Index: po/te.po
===================================================================
Index: po/en_GB.po
===================================================================
Index: po/si.po
===================================================================
Index: po/cy.po
===================================================================
Index: po/gu.po
===================================================================
Index: po/hu.po
===================================================================
Index: po/vi.po
===================================================================
Index: po/hy.po
===================================================================
Index: po/my.po
===================================================================
Index: po/ru.po
===================================================================
Index: po/id.po
===================================================================
Index: po/el.po
===================================================================
Index: po/ml.po
===================================================================
Index: po/et.po
===================================================================
Index: po/nl.po
===================================================================
Index: po/pl.po
===================================================================
Index: po/it.po
===================================================================
Index: po/sl.po
===================================================================
Index: po/pt.po
===================================================================
Index: po/bg.po
===================================================================
Index: system-config-printer.glade
===================================================================
--- system-config-printer.glade	(.../tags/0.7.74.2)	(revision 1574)
+++ system-config-printer.glade	(.../branches/0.7.74.x)	(revision 1574)
@@ -432,6 +432,40 @@
 			      </child>
 
 			      <child>
+				<widget class="GtkAlignment" id="alignment78">
+				  <property name="visible">True</property>
+				  <property name="xalign">0.5</property>
+				  <property name="yalign">0.5</property>
+				  <property name="xscale">1</property>
+				  <property name="yscale">1</property>
+				  <property name="top_padding">0</property>
+				  <property name="bottom_padding">0</property>
+				  <property name="left_padding">12</property>
+				  <property name="right_padding">0</property>
+
+				  <child>
+				    <widget class="GtkCheckButton" id="chkServerShareAny">
+				      <property name="visible">True</property>
+				      <property name="can_focus">True</property>
+				      <property name="label" translatable="yes">Allow printing from the Internet</property>
+				      <property name="use_underline">True</property>
+				      <property name="relief">GTK_RELIEF_NORMAL</property>
+				      <property name="focus_on_click">True</property>
+				      <property name="active">False</property>
+				      <property name="inconsistent">False</property>
+				      <property name="draw_indicator">True</property>
+				      <signal name="toggled" handler="on_server_changed" last_modification_time="Tue, 04 Sep 2007 09:46:49 GMT"/>
+				    </widget>
+				  </child>
+				</widget>
+				<packing>
+				  <property name="padding">0</property>
+				  <property name="expand">False</property>
+				  <property name="fill">False</property>
+				</packing>
+			      </child>
+
+			      <child>
 				<widget class="GtkCheckButton" id="chkServerRemoteAdmin">
 				  <property name="visible">True</property>
 				  <property name="can_focus">True</property>
Index: ChangeLog
===================================================================
--- ChangeLog	(.../tags/0.7.74.2)	(revision 1574)
+++ ChangeLog	(.../branches/0.7.74.x)	(revision 1574)
@@ -1,3 +1,26 @@
+2007-09-25  Tim Waugh  <twaugh@redhat.com>
+
+	* ppds.py (getDriverType): Fix foomatic recommended driver
+	discovery (requires CUPS patch to prevent '(recommended)' string
+	getting stripped).
+	(PPDs.getDriverTypeWithBias): Let getDriverType have access to the
+	PPDs object.
+
+2007-09-04  Tim Waugh  <twaugh@redhat.com>
+
+	* system-config-printer.py (GUI.setConnected): Set sensitivity for
+	new check-box.
+	(GUI.fillServerTab): Show setting for new check-box.
+	(GUI.on_server_changed): Set sensitivity for new check-box.
+	(GUI.save_serversettings): Set server setting from check-box.
+
+	* system-config-printer.glade: New check-box for 'Allow printing
+	from the Internet'.
+
+2007-09-21  Tim Waugh  <twaugh@redhat.com>
+
+	* po/ko.po: Updated (from 0.7.32.x POT; needs further updating).
+
 2007-09-19  Tim Waugh  <twaugh@redhat.com>
 
 	* configure.in: Version 0.7.74.2.
Index: ppds.py
===================================================================
--- ppds.py	(.../tags/0.7.74.2)	(revision 1574)
+++ ppds.py	(.../branches/0.7.74.x)	(revision 1574)
@@ -112,7 +112,7 @@
 DRIVER_TYPE_CUPS = 80
 DRIVER_TYPE_FOOMATIC_GENERIC = 90
 DRIVER_DOES_NOT_WORK = 999
-def getDriverType (ppdname):
+def getDriverType (ppdname, ppds=None):
     """Decides which of the above types ppdname is."""
     if ppdname.startswith ("gutenprint"):
         if ppdname.find ("/simple/") != -1:
@@ -128,7 +128,9 @@
         # Foomatic (generated) -- but which driver?
         if ppdname.find ("Generic")!= -1:
             return DRIVER_TYPE_FOOMATIC_GENERIC
-        if ppdname.find ("(recommended)")!= -1:
+        if (ppds != None and
+            ppds.getInfoFromPPDName (ppdname).\
+            get ('ppd-make-and-model', '').find ("(recommended)") != -1):
             return DRIVER_TYPE_FOOMATIC_RECOMMENDED
         if ppdname.find ("-Postscript")!= -1:
             return DRIVER_TYPE_FOOMATIC_PS
@@ -234,7 +236,7 @@
         make_model = dict['ppd-make-and-model']
         mfg, mdl = ppdMakeModelSplit (make_model)
         def getDriverTypeWithBias (x, mfg):
-            t = getDriverType (x)
+            t = getDriverType (x, ppds=self)
             if mfg == "HP":
                 if t == DRIVER_TYPE_FOOMATIC_HPIJS:
                     # Prefer HPIJS for HP devices.
Index: system-config-printer.py
===================================================================
--- system-config-printer.py	(.../tags/0.7.74.2)	(revision 1574)
+++ system-config-printer.py	(.../branches/0.7.74.x)	(revision 1574)
@@ -74,6 +74,12 @@
 ready_cursor = gtk.gdk.Cursor(gtk.gdk.LEFT_PTR)
 ellipsis = unichr(0x2026)
 
+try:
+    try_CUPS_SERVER_REMOTE_ANY = cups.CUPS_SERVER_REMOTE_ANY
+except AttributeError:
+    # cups module was compiled with CUPS < 1.3
+    try_CUPS_SERVER_REMOTE_ANY = "_remote_any"
+
 def nonfatalException ():
     print "Caught non-fatal exception.  Traceback:"
     (type, value, tb) = sys.exc_info ()
@@ -135,6 +141,7 @@
                         "btnApply", "btnRevert", "btnConflict",
 
                         "chkServerBrowse", "chkServerShare",
+                        "chkServerShareAny",
                         "chkServerRemoteAdmin", "chkServerAllowCancelAll",
                         "chkServerLogDebug",
 
@@ -607,6 +614,9 @@
                        self.chkServerLogDebug):
             widget.set_sensitive(connected)
 
+        sharing = self.chkServerShare.get_active ()
+        self.chkServerShareAny.set_sensitive (sharing)
+
         try:
             del self.server_settings
         except:
@@ -3625,15 +3635,17 @@
         for widget, setting in [
             (self.chkServerBrowse, cups.CUPS_SERVER_REMOTE_PRINTERS),
             (self.chkServerShare, cups.CUPS_SERVER_SHARE_PRINTERS),
+            (self.chkServerShareAny, try_CUPS_SERVER_REMOTE_ANY),
             (self.chkServerRemoteAdmin, cups.CUPS_SERVER_REMOTE_ADMIN),
             (self.chkServerAllowCancelAll, cups.CUPS_SERVER_USER_CANCEL_ANY),
             (self.chkServerLogDebug, cups.CUPS_SERVER_DEBUG_LOGGING),]:
             widget.set_data("setting", setting)
             if self.server_settings.has_key(setting):
                 widget.set_active(int(self.server_settings[setting]))
-                widget.show()
+                widget.set_sensitive(True)
             else:
-                widget.hide()
+                widget.set_active(False)
+                widget.set_sensitive(False)
         self.setDataButtonState()
         
     def on_server_changed(self, widget):
@@ -3642,6 +3654,11 @@
             self.changed.discard(widget)
         else:
             self.changed.add(widget)
+
+        sharing = self.chkServerShare.get_active ()
+        self.chkServerShareAny.set_sensitive (
+            sharing and self.server_settings.has_key(try_CUPS_SERVER_REMOTE_ANY))
+
         self.setDataButtonState()
 
     def save_serversettings(self):
@@ -3649,6 +3666,7 @@
         for widget, setting in [
             (self.chkServerBrowse, cups.CUPS_SERVER_REMOTE_PRINTERS),
             (self.chkServerShare, cups.CUPS_SERVER_SHARE_PRINTERS),
+            (self.chkServerShareAny, try_CUPS_SERVER_REMOTE_ANY),
             (self.chkServerRemoteAdmin, cups.CUPS_SERVER_REMOTE_ADMIN),
             (self.chkServerAllowCancelAll, cups.CUPS_SERVER_USER_CANCEL_ANY),
             (self.chkServerLogDebug, cups.CUPS_SERVER_DEBUG_LOGGING),]:
