diff -up system-config-printer-1.1.15/system-config-printer.py.selection-browse system-config-printer-1.1.15/system-config-printer.py
--- system-config-printer-1.1.15/system-config-printer.py.selection-browse	2009-12-16 16:54:39.000000000 +0100
+++ system-config-printer-1.1.15/system-config-printer.py	2009-12-18 11:46:36.000000000 +0100
@@ -646,6 +646,7 @@ class GUI(GtkGUI, monitor.Watcher):
         # setup some lists
         m = gtk.SELECTION_MULTIPLE
         s = gtk.SELECTION_SINGLE
+        b = gtk.SELECTION_BROWSE
         for name, treeview, selection_mode in (
             (_("Members of this class"), self.tvClassMembers, m),
             (_("Others"), self.tvClassNotMembers, m),
@@ -656,7 +657,7 @@ class GUI(GtkGUI, monitor.Watcher):
             (_("Makes"), np.tvNPMakes,s),
             (_("Models"), np.tvNPModels,s),
             (_("Drivers"), np.tvNPDrivers,s),
-            (_("Downloadable Drivers"), np.tvNPDownloadableDrivers,s),
+            (_("Downloadable Drivers"), np.tvNPDownloadableDrivers, b),
             (_("Users"), self.tvPUsers, m),
             ):
 
