diff -up system-config-printer-1.2.0/check-device-ids.py.cdi-search-harder system-config-printer-1.2.0/check-device-ids.py
--- system-config-printer-1.2.0/check-device-ids.py.cdi-search-harder	2010-03-25 16:33:07.312234926 +0000
+++ system-config-printer-1.2.0/check-device-ids.py	2010-03-25 16:33:07.409361225 +0000
@@ -195,6 +195,7 @@ for device, attrs in devices.iteritems (
         print "%s   (No drivers)" % more
 
     (mfr, mdl) = ppdMakeModelSplit (make_and_model)
+    matches = set (ppds.getInfoFromModel (mfr, mdl))
     mfrl = mfr.lower ()
     mdls = None
     for make in makes:
@@ -204,9 +205,10 @@ for device, attrs in devices.iteritems (
     if mdls:
         (s, bestmatches) = ppds._findBestMatchPPDs (mdls, id_fields['MDL'])
         if s == ppds.STATUS_SUCCESS:
-            missing = set (bestmatches) - set (drivers)
-            for each in missing:
-                print "%s       MISSING  %s [%s]" % (more, each,
-                                                     driver_uri_to_pkg (each))
+            matches = matches.union (set (bestmatches))
+
+    missing = set (matches) - set (drivers)
+    for each in missing:
+        print "%s       MISSING  %s" % (more, each)
 
     i += 1
