diff -up system-config-printer-1.1.90/ui/PrinterPropertiesDialog.glade.spinbuttons system-config-printer-1.1.90/ui/PrinterPropertiesDialog.glade
--- system-config-printer-1.1.90/ui/PrinterPropertiesDialog.glade.spinbuttons	2010-01-19 11:18:46.000000000 +0000
+++ system-config-printer-1.1.90/ui/PrinterPropertiesDialog.glade	2010-01-21 14:45:53.270150394 +0000
@@ -1313,6 +1313,7 @@ See server settings&lt;/i&gt;</property>
                                                       <object class="GtkSpinButton" id="sbJOBrightness">
                                                         <property name="visible">True</property>
                                                         <property name="can_focus">True</property>
+                                                        <property name="adjustment">adjustment3</property>
                                                         <property name="climb_rate">1</property>
                                                         <accessibility>
                                                           <relation type="labelled-by" target="label168"/>
@@ -1415,6 +1416,7 @@ See server settings&lt;/i&gt;</property>
                                                   <object class="GtkSpinButton" id="sbJOJobPriority">
                                                     <property name="visible">True</property>
                                                     <property name="can_focus">True</property>
+                                                    <property name="adjustment">adjustment4</property>
                                                     <property name="climb_rate">1</property>
                                                     <accessibility>
                                                       <relation type="labelled-by" target="label177"/>
@@ -1671,7 +1673,9 @@ See server settings&lt;/i&gt;</property>
                                           <object class="GtkSpinButton" id="sbJOCopies">
                                             <property name="visible">True</property>
                                             <property name="can_focus">True</property>
+                                            <property name="adjustment">adjustment1</property>
                                             <property name="climb_rate">1</property>
+                                            <property name="numeric">True</property>
                                             <accessibility>
                                               <relation type="labelled-by" target="label138"/>
                                             </accessibility>
@@ -1835,6 +1839,7 @@ See server settings&lt;/i&gt;</property>
                                               <object class="GtkSpinButton" id="sbJOScaling">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">True</property>
+                                                <property name="adjustment">adjustment2</property>
                                                 <property name="climb_rate">1</property>
                                                 <accessibility>
                                                   <relation type="labelled-by" target="label144"/>
@@ -1916,6 +1921,7 @@ See server settings&lt;/i&gt;</property>
                                                       <object class="GtkSpinButton" id="sbJOSaturation">
                                                         <property name="visible">True</property>
                                                         <property name="can_focus">True</property>
+                                                        <property name="adjustment">adjustment5</property>
                                                         <property name="climb_rate">1</property>
                                                         <accessibility>
                                                           <relation type="labelled-by" target="label170"/>
@@ -1985,6 +1991,7 @@ See server settings&lt;/i&gt;</property>
                                                       <object class="GtkSpinButton" id="sbJOHue">
                                                         <property name="visible">True</property>
                                                         <property name="can_focus">True</property>
+                                                        <property name="adjustment">adjustment6</property>
                                                         <property name="climb_rate">1</property>
                                                         <accessibility>
                                                           <relation type="labelled-by" target="label172"/>
@@ -2039,6 +2046,7 @@ See server settings&lt;/i&gt;</property>
                                                   <object class="GtkSpinButton" id="sbJOGamma">
                                                     <property name="visible">True</property>
                                                     <property name="can_focus">True</property>
+                                                    <property name="adjustment">adjustment7</property>
                                                     <property name="climb_rate">1</property>
                                                     <accessibility>
                                                       <relation type="labelled-by" target="label174"/>
@@ -2199,6 +2207,7 @@ See server settings&lt;/i&gt;</property>
                                                   <object class="GtkSpinButton" id="sbJOPageRight">
                                                     <property name="visible">True</property>
                                                     <property name="can_focus">True</property>
+                                                    <property name="adjustment">adjustment11</property>
                                                     <property name="climb_rate">1</property>
                                                     <accessibility>
                                                       <relation type="labelled-by" target="label156"/>
@@ -2241,6 +2250,7 @@ See server settings&lt;/i&gt;</property>
                                               <object class="GtkSpinButton" id="sbJOCpi">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">True</property>
+                                                <property name="adjustment">adjustment8</property>
                                                 <property name="climb_rate">1</property>
                                                 <property name="digits">2</property>
                                                 <accessibility>
@@ -2280,6 +2290,7 @@ See server settings&lt;/i&gt;</property>
                                               <object class="GtkSpinButton" id="sbJOLpi">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">True</property>
+                                                <property name="adjustment">adjustment9</property>
                                                 <property name="climb_rate">1</property>
                                                 <property name="digits">2</property>
                                                 <accessibility>
@@ -2337,6 +2348,7 @@ See server settings&lt;/i&gt;</property>
                                               <object class="GtkSpinButton" id="sbJOPageLeft">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">True</property>
+                                                <property name="adjustment">adjustment10</property>
                                                 <property name="climb_rate">1</property>
                                                 <accessibility>
                                                   <relation type="labelled-by" target="label154"/>
@@ -2465,6 +2477,7 @@ See server settings&lt;/i&gt;</property>
                                                   <object class="GtkSpinButton" id="sbJOColumns">
                                                     <property name="visible">True</property>
                                                     <property name="can_focus">True</property>
+                                                    <property name="adjustment">adjustment14</property>
                                                     <property name="climb_rate">1</property>
                                                     <property name="numeric">True</property>
                                                     <accessibility>
@@ -2559,6 +2572,7 @@ See server settings&lt;/i&gt;</property>
                                               <object class="GtkSpinButton" id="sbJOPageTop">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">True</property>
+                                                <property name="adjustment">adjustment12</property>
                                                 <property name="climb_rate">1</property>
                                                 <accessibility>
                                                   <relation type="labelled-by" target="label158"/>
@@ -2615,6 +2629,7 @@ See server settings&lt;/i&gt;</property>
                                               <object class="GtkSpinButton" id="sbJOPageBottom">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">True</property>
+                                                <property name="adjustment">adjustment13</property>
                                                 <property name="climb_rate">1</property>
                                                 <accessibility>
                                                   <relation type="labelled-by" target="label160"/>
@@ -3244,4 +3259,103 @@ See server settings&lt;/i&gt;</property>
       <action-widget response="-6">btnPrinterPropertiesClose</action-widget>
     </action-widgets>
   </object>
+  <object class="GtkAdjustment" id="adjustment1">
+    <property name="value">1</property>
+    <property name="upper">1000</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment2">
+    <property name="value">100</property>
+    <property name="upper">1000</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment3">
+    <property name="value">100</property>
+    <property name="upper">1000</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment4">
+    <property name="value">50</property>
+    <property name="lower">1</property>
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment5">
+    <property name="value">100</property>
+    <property name="upper">1000</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment6">
+    <property name="lower">-180</property>
+    <property name="upper">180</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment7">
+    <property name="value">1000</property>
+    <property name="lower">1</property>
+    <property name="upper">10000</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment8">
+    <property name="value">10</property>
+    <property name="lower">1</property>
+    <property name="upper">100</property>
+    <property name="step_increment">0.10000000000000001</property>
+    <property name="page_increment">1</property>
+    <property name="page_size">1</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment9">
+    <property name="value">6</property>
+    <property name="lower">1</property>
+    <property name="upper">100</property>
+    <property name="step_increment">0.10000000000000001</property>
+    <property name="page_increment">1</property>
+    <property name="page_size">1</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment10">
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment11">
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment12">
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment13">
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+    <property name="page_size">10</property>
+  </object>
+  <object class="GtkAdjustment" id="adjustment14">
+    <property name="value">1</property>
+    <property name="lower">1</property>
+    <property name="upper">4</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">1</property>
+    <property name="page_size">1</property>
+  </object>
 </interface>
