diff -up system-config-printer-1.1.12/ppdippstr.py.ppdippstr system-config-printer-1.1.12/ppdippstr.py
--- system-config-printer-1.1.12/ppdippstr.py.ppdippstr	2009-08-25 15:01:41.000000000 +0100
+++ system-config-printer-1.1.12/ppdippstr.py	2009-08-28 16:47:05.964949870 +0100
@@ -21,6 +21,11 @@
 
 from gettext import gettext as _
 
+printer_error_policy = dict()
+printer_op_policy = dict()
+job_sheets = dict()
+ppd = dict()
+
 class TranslationDict:
     STR = {}
 
@@ -30,104 +35,111 @@ class TranslationDict:
     def get (self, str):
         return self.STR.get (str, str)
 
-## IPP strings
+def init ():
+    ## IPP strings
 
-# Names of printer error policies
-printer_error_policy = TranslationDict ({
-        "abort-job": _("Abort job"),
-        "retry-job": _("Retry job"),
-        "stop-printer": _("Stop printer")
-        })
+    # Names of printer error policies
+    global printer_error_policy
+    printer_error_policy = TranslationDict ({
+            "abort-job": _("Abort job"),
+            "retry-job": _("Retry job"),
+            "stop-printer": _("Stop printer")
+            })
     
-# Names of printer operation policies
-printer_op_policy = TranslationDict ({
-        "default": _("Default behavior"),
-        "authenticated": _("Authenticated")
-        })
-
-# Names of banner pages.
-job_sheets = TranslationDict ({
-        "none": _("None"),
-        "classified": _("Classified"),
-        "confidential": _("Confidential"),
-        "secret": _("Secret"),
-        "standard": _("Standard"),
-        "topsecret": _("Top secret"),
-        "unclassified": _("Unclassified")
-        })
-
-## Common PPD strings
-
-# Foomatic strings
-
-# These are PPD option and group names and values.
-ppd = TranslationDict ({
-        "General": _("General"),
-
-        # HPIJS options
-        "Printout Mode": _("Printout mode"),
-        "Draft (auto-detect paper type)": _("Draft (auto-detect-paper type)"),
-        "Draft Grayscale (auto-detect paper type)":
-            _("Draft grayscale (auto-detect-paper type)"),
-        "Normal (auto-detect paper type)": _("Normal (auto-detect-paper type)"),
-        "Normal Grayscale (auto-detect paper type)":
-            _("Normal grayscale (auto-detect-paper type)"),
-        "High Quality (auto-detect paper type)":
-            _("High quality (auto-detect-paper type)"),
-        "High Quality Grayscale (auto-detect paper type)":
-            _("High quality grayscale (auto-detect-paper type)"),
-        "Photo (on photo paper)": _("Photo (on photo paper)"),
-        "Best Quality (color on photo paper)":
-            _("Best quality (color on photo paper)"),
-        "Normal Quality (color on photo paper)":
-            _("Normal quality (color on photo paper)"),
-
-        "Media Source": _("Media source"),
-        "Printer default": _("Printer default"),
-        "Photo Tray": _("Photo tray"),
-        "Upper Tray": _("Upper tray"),
-        "Lower Tray": _("Lower tray"),
-        "CD or DVD Tray": _("CD or DVD tray"),
-        "Envelope Feeder": _("Envelope feeder"),
-        "Large Capacity Tray": _("Large capacity tray"),
-        "Manual Feeder": _("Manual feeder"),
-        "Multi Purpose Tray": _("Multi-purpose tray"),
-
-        "Page Size": _("Page size"),
-        "Custom": _("Custom"),
-        "Photo or 4x6 inch index card": _("Photo or 4x6 inch index card"),
-        "Photo or 5x7 inch index card": _("Photo or 5x7 inch index card"),
-        "Photo with tear-off tab": _("Photo with tear-off tab"),
-        "3x5 inch index card": _("3x5 inch index card"),
-        "5x8 inch index card": _("5x8 inch index card"),
-        "A6 with tear-off tab": _("A6 with tear-off tab"),
-        "CD or DVD 80 mm": _("CD or DVD 80mm"),
-        "CD or DVD 120 mm": _("CD or DVD 120mm"),
-
-        "Double-Sided Printing": _("Double-sided printing"),
-        "Long Edge (Standard)": _("Long edge (standard)"),
-        "Short Edge (Flip)": _("Short edge (flip)"),
-        "Off": _("Off"),
-
-        "Resolution, Quality, Ink Type, Media Type":
-            _("Resolution, quality, ink type, media type"),
-        "Controlled by 'Printout Mode'": _("Controlled by 'Printout mode'"),
-        "300 dpi, Color, Black + Color Cartr.":
-            _("300 dpi, color, black + color cartridge"),
-        "300 dpi, Draft, Color, Black + Color Cartr.":
-            _("300 dpi, draft, color, black + color cartridge"),
-        "300 dpi, Draft, Grayscale, Black + Color Cartr.":
-            _("300 dpi, draft, grayscale, black + color cartridge"),
-        "300 dpi, Grayscale, Black + Color Cartr.":
-            _("300 dpi, grayscale, black + color cartridge"),
-        "600 dpi, Color, Black + Color Cartr.":
-            _("600 dpi, color, black + color cartridge"),
-        "600 dpi, Grayscale, Black + Color Cartr.":
-            _("600 dpi, grayscale, black + color cartridge"),
-        "600 dpi, Photo, Black + Color Cartr., Photo Paper":
-            _("600 dpi, photo, black + color cartridge, photo paper"),
-        "600 dpi, Color, Black + Color Cartr., Photo Paper, Normal":
-            _("600 dpi, color, black + color cartridge, photo paper, normal"),
-        "1200 dpi, Photo, Black + Color Cartr., Photo Paper":
-            _("1200 dpi, photo, black + color cartridge, photo paper"),
-        })
+    # Names of printer operation policies
+    global printer_op_policy
+    printer_op_policy = TranslationDict ({
+            "default": _("Default behavior"),
+            "authenticated": _("Authenticated")
+            })
+
+    # Names of banner pages.
+    global job_sheets
+    job_sheets = TranslationDict ({
+            "none": _("None"),
+            "classified": _("Classified"),
+            "confidential": _("Confidential"),
+            "secret": _("Secret"),
+            "standard": _("Standard"),
+            "topsecret": _("Top secret"),
+            "unclassified": _("Unclassified")
+            })
+
+    ## Common PPD strings
+
+    # Foomatic strings
+
+    # These are PPD option and group names and values.
+    global ppd
+    ppd = TranslationDict ({
+            "General": _("General"),
+
+            # HPIJS options
+            "Printout Mode": _("Printout mode"),
+            "Draft (auto-detect paper type)":
+                _("Draft (auto-detect-paper type)"),
+            "Draft Grayscale (auto-detect paper type)":
+                _("Draft grayscale (auto-detect-paper type)"),
+            "Normal (auto-detect paper type)":
+                _("Normal (auto-detect-paper type)"),
+            "Normal Grayscale (auto-detect paper type)":
+                _("Normal grayscale (auto-detect-paper type)"),
+            "High Quality (auto-detect paper type)":
+                _("High quality (auto-detect-paper type)"),
+            "High Quality Grayscale (auto-detect paper type)":
+                _("High quality grayscale (auto-detect-paper type)"),
+            "Photo (on photo paper)": _("Photo (on photo paper)"),
+            "Best Quality (color on photo paper)":
+                _("Best quality (color on photo paper)"),
+            "Normal Quality (color on photo paper)":
+                _("Normal quality (color on photo paper)"),
+
+            "Media Source": _("Media source"),
+            "Printer default": _("Printer default"),
+            "Photo Tray": _("Photo tray"),
+            "Upper Tray": _("Upper tray"),
+            "Lower Tray": _("Lower tray"),
+            "CD or DVD Tray": _("CD or DVD tray"),
+            "Envelope Feeder": _("Envelope feeder"),
+            "Large Capacity Tray": _("Large capacity tray"),
+            "Manual Feeder": _("Manual feeder"),
+            "Multi Purpose Tray": _("Multi-purpose tray"),
+
+            "Page Size": _("Page size"),
+            "Custom": _("Custom"),
+            "Photo or 4x6 inch index card": _("Photo or 4x6 inch index card"),
+            "Photo or 5x7 inch index card": _("Photo or 5x7 inch index card"),
+            "Photo with tear-off tab": _("Photo with tear-off tab"),
+            "3x5 inch index card": _("3x5 inch index card"),
+            "5x8 inch index card": _("5x8 inch index card"),
+            "A6 with tear-off tab": _("A6 with tear-off tab"),
+            "CD or DVD 80 mm": _("CD or DVD 80mm"),
+            "CD or DVD 120 mm": _("CD or DVD 120mm"),
+
+            "Double-Sided Printing": _("Double-sided printing"),
+            "Long Edge (Standard)": _("Long edge (standard)"),
+            "Short Edge (Flip)": _("Short edge (flip)"),
+            "Off": _("Off"),
+
+            "Resolution, Quality, Ink Type, Media Type":
+                _("Resolution, quality, ink type, media type"),
+            "Controlled by 'Printout Mode'": _("Controlled by 'Printout mode'"),
+            "300 dpi, Color, Black + Color Cartr.":
+                _("300 dpi, color, black + color cartridge"),
+            "300 dpi, Draft, Color, Black + Color Cartr.":
+                _("300 dpi, draft, color, black + color cartridge"),
+            "300 dpi, Draft, Grayscale, Black + Color Cartr.":
+                _("300 dpi, draft, grayscale, black + color cartridge"),
+            "300 dpi, Grayscale, Black + Color Cartr.":
+                _("300 dpi, grayscale, black + color cartridge"),
+            "600 dpi, Color, Black + Color Cartr.":
+                _("600 dpi, color, black + color cartridge"),
+            "600 dpi, Grayscale, Black + Color Cartr.":
+                _("600 dpi, grayscale, black + color cartridge"),
+            "600 dpi, Photo, Black + Color Cartr., Photo Paper":
+                _("600 dpi, photo, black + color cartridge, photo paper"),
+            "600 dpi, Color, Black + Color Cartr., Photo Paper, Normal":
+                _("600 dpi, color, black + color cartridge, photo paper, normal"),
+            "1200 dpi, Photo, Black + Color Cartr., Photo Paper":
+                _("1200 dpi, photo, black + color cartridge, photo paper"),
+            })
diff -up system-config-printer-1.1.12/system-config-printer.py.ppdippstr system-config-printer-1.1.12/system-config-printer.py
--- system-config-printer-1.1.12/system-config-printer.py.ppdippstr	2009-08-28 16:46:47.922947897 +0100
+++ system-config-printer-1.1.12/system-config-printer.py	2009-08-28 16:47:06.032823573 +0100
@@ -129,6 +129,7 @@ gettext.bindtextdomain (domain, config.l
 gtk.glade.textdomain (domain)
 gtk.glade.bindtextdomain (domain, config.localedir)
 import ppdippstr
+ppdippstr.init ()
 pkgdata = config.pkgdatadir
 iconpath = os.path.join (pkgdata, 'icons/')
 sys.path.append (pkgdata)
