diff -up system-config-printer-1.1.8/system-config-printer.py.properties-cancel system-config-printer-1.1.8/system-config-printer.py
--- system-config-printer-1.1.8/system-config-printer.py.properties-cancel	2009-06-25 17:50:16.256328643 +0100
+++ system-config-printer-1.1.8/system-config-printer.py	2009-06-25 17:50:37.650328968 +0100
@@ -1045,7 +1045,7 @@ class GUI(GtkGUI, monitor.Watcher):
 
         if ((response == gtk.RESPONSE_OK and not success) or
             response == gtk.RESPONSE_CANCEL):
-            del self.printer
+            self.printer = None
             dialog.hide ()
 
     def dests_iconview_selection_changed (self, iconview):
