diff -up system-config-printer-1.1.13/system-config-printer.py.notification-traceback system-config-printer-1.1.13/system-config-printer.py
--- system-config-printer-1.1.13/system-config-printer.py.notification-traceback	2009-10-30 12:36:24.901645742 +0000
+++ system-config-printer-1.1.13/system-config-printer.py	2009-10-30 12:36:34.610518996 +0000
@@ -3494,8 +3494,11 @@ class GUI(GtkGUI, monitor.Watcher):
             self.dests_iconview_selection_changed (self.dests_iconview)
             gobject.idle_add (deferred_refresh)
             if self.PrinterPropertiesDialog.get_property('visible'):
-                self.printer.getAttributes ()
-                self.updatePrinterProperties ()
+                try:
+                    self.printer.getAttributes ()
+                    self.updatePrinterProperties ()
+                except cups.IPPError:
+                    pass
 
         gtk.gdk.threads_leave ()
 
@@ -3507,8 +3510,11 @@ class GUI(GtkGUI, monitor.Watcher):
         monitor.Watcher.state_reason_added (self, mon, reason)
         gtk.gdk.threads_enter ()
         if self.PrinterPropertiesDialog.get_property('visible'):
-            self.printer.getAttributes ()
-            self.updatePrinterProperties ()
+            try:
+                self.printer.getAttributes ()
+                self.updatePrinterProperties ()
+            except cups.IPPError:
+                pass
 
         gtk.gdk.threads_leave ()
 
@@ -3516,8 +3522,11 @@ class GUI(GtkGUI, monitor.Watcher):
         monitor.Watcher.state_reason_removed (self, mon, reason)
         gtk.gdk.threads_enter ()
         if self.PrinterPropertiesDialog.get_property('visible'):
-            self.printer.getAttributes ()
-            self.updatePrinterProperties ()
+            try:
+                self.printer.getAttributes ()
+                self.updatePrinterProperties ()
+            except cups.IPPError:
+                pass
 
         gtk.gdk.threads_leave ()
 
