diff -up system-config-printer-1.2.0/check-device-ids.py.cdi-make-and-model system-config-printer-1.2.0/check-device-ids.py
--- system-config-printer-1.2.0/check-device-ids.py.cdi-make-and-model	2010-03-27 10:16:04.371239756 +0000
+++ system-config-printer-1.2.0/check-device-ids.py	2010-03-27 10:16:14.458240137 +0000
@@ -210,7 +210,7 @@ for device, attrs in devices.iteritems (
             mdls = ppds.makes[make]
             break
     if mdls:
-        (s, bestmatches) = ppds._findBestMatchPPDs (mdls, id_fields['MDL'])
+        (s, bestmatches) = ppds._findBestMatchPPDs (mdls, mdl)
         if s == ppds.STATUS_SUCCESS:
             matches = matches.union (set (bestmatches))
 
