diff -up system-config-printer-1.4.4/scp-dbus-service.py.openprinting-freeze system-config-printer-1.4.4/scp-dbus-service.py
--- system-config-printer-1.4.4/scp-dbus-service.py.openprinting-freeze	2014-01-28 16:02:43.000000000 +0000
+++ system-config-printer-1.4.4/scp-dbus-service.py	2014-05-01 13:00:04.048257135 +0100
@@ -194,34 +194,6 @@ class GetBestDriversRequest:
                                                           fit=fit)
             ppdname = ppdnamelist[0]
             status = fit[ppdname]
-
-            try:
-                if status != "exact" and not self.download_tried:
-                    self.download_tried = True
-                    self.loop = GObject.MainLoop ()
-                    self.dialog = newprinter.NewPrinterGUI()
-                    self.dialog.NewPrinterWindow.set_modal (False)
-                    self.handles = \
-                        [self.dialog.connect ('dialog-canceled',
-                                              self.on_dialog_canceled),
-                         self.dialog.connect ('driver-download-checked',
-                                              self.on_driver_download_checked)]
-                    if self.dialog.init ('download_driver',
-                                         devid=self.device_id):
-                        self.loop.run()
-                    for handle in self.handles:
-                        self.dialog.disconnect (handle)
-                    if len(self.installed_files) > 0:
-                        debugprint ("GetBestDrivers request: Re-fetch PPDs after driver download")
-                        g_ppds = FetchedPPDs (self.cupsconn, self.language)
-                        self._signals.append (g_ppds.connect ('ready', self._ppds_ready))
-                        self._signals.append (g_ppds.connect ('error', self._ppds_error))
-                        g_ppds.run ()
-                        return
-            except:
-                # Ignore driver download if packages needed for the GUI are not
-                # installed or if no windows can be opened
-                pass
             g_killtimer.remove_hold ()
             self.reply_handler (map (lambda x: (x, fit[x]), ppdnamelist))
         except Exception as e:
