diff -up system-config-printer-1.1.13/system-config-printer.py.cancel-traceback system-config-printer-1.1.13/system-config-printer.py
--- system-config-printer-1.1.13/system-config-printer.py.cancel-traceback	2009-09-18 13:10:10.962438215 +0100
+++ system-config-printer-1.1.13/system-config-printer.py	2009-09-18 13:10:20.041314240 +0100
@@ -3805,6 +3805,7 @@ class NewPrinterGUI(GtkGUI):
         self.options = {} # keyword -> Option object
         self.changed = set()
         self.conflicts = set()
+        self.fetchDevices_op = None
 
         combobox = self.cmbNPDownloadableDriverFoundPrinters
         combobox.set_model (gtk.ListStore (str, str))
