diff -up system-config-printer-1.5.9/Makefile.am.bz1419175 system-config-printer-1.5.9/Makefile.am
--- system-config-printer-1.5.9/Makefile.am.bz1419175	2017-03-08 15:00:02.434967635 +0100
+++ system-config-printer-1.5.9/Makefile.am	2017-03-08 15:01:06.992438164 +0100
@@ -52,7 +52,7 @@ cupshelpers/config.py: cupshelpers/confi
 	  -e "s|\@cupsserverbindir\@|$(cupsserverbindir)|" \
 	  $< > $@
 
-dbus/org.fedoraproject.Config.Printing.service: dbus/org.fedoraproject.Config.Printing.service.in
+dbus/org.fedoraproject.Config.Printing.service: dbus/org.fedoraproject.Config.Printing.service.in $(top_builddir)/config.status Makefile
 	sed \
 	-e "s|\@bindir\@|$(bindir)|" \
 	$< >$@
@@ -222,7 +222,7 @@ dbusinterfaces_DATA =				\
 	dbus/org.fedoraproject.Config.Printing.xml
 dbusinterfacesdir = $(datadir)/dbus-1/interfaces/
 
-dbusservices_DATA =				\
+nodist_dbusservices_DATA =				\
 	dbus/org.fedoraproject.Config.Printing.service
 dbusservicesdir = $(datadir)/dbus-1/services/
 
